js/indicators/regressions.src.js

```diff
@@ -0,0 +1,387 @@
+/**
+ * (c) 2010-2018 Kamil Kulig
+ *
+ * License: www.highcharts.com/license
+ */
+
+'use strict';
+import H from '../parts/Globals.js';
+import '../parts/Utilities.js';
+
+var seriesType = H.seriesType,
+    reduce = H.reduce,
+    map = H.map,
+    isArray = H.isArray;
+
+/**
+ * Linear regression series type.
+ *
+ * @constructor seriesTypes.linearRegression
+ * @augments seriesTypes.line
+ */
+seriesType('linearRegression', 'sma',
+    /**
+     * Linear regression indicator. This series requires `linkedTo`
+     * option to be set.
+     *
+     * @extends plotOptions.sma
+     * @product highstock
+     * @sample {highstock} stock/indicators/linear-regression
+     *                     Linear regression indicator
+     * @since 7.0.0
+     * @optionparent plotOptions.linearRegression
+     */
+    {
+        name: 'Linear Regression Indicator',
```

Comment:
By default, shouldn't LRI include `period` in the name? For example `Linear Regression Indicator (2)` ? Check ZigZag indicator for example (`nameBase` should be enough, I think).